runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,297 @@
+		{
+			Widget viewportWidget = client.getWidget(viewport.getViewport());
+			if (viewportWidget != null && !viewportWidget.isHidden())
+			{
+				curViewport = viewport;
+				curWidget = viewportWidget;
+				break;
+			}
+		}
+
+		if (curViewport == null || curWidget.isHidden())
+		{
+			return null;
+		}
+
+		final Point offsetLeft = curViewport.getOffsetLeft();
+		final Point offsetRight = curViewport.getOffsetRight();
+		final Point location = curWidget.getCanvasLocation();
+		int offsetPrayerX = (location.getX() - offsetRight.getX());
+		int offsetHealthX = (location.getX() - offsetLeft.getX());
+		int offsetPrayerY = (location.getY() - offsetRight.getY());
+		int offsetHealthY = (location.getY() - offsetLeft.getY());
+		int height = curViewport == Viewport.RESIZED_BOTTOM ? RESIZED_BOTTOM_HEIGHT : HEIGHT;
+
+		if (curViewport == Viewport.RESIZED_BOTTOM)
+		{
+			offsetPrayerX = (location.getX() + RESIZED_BOTTOM_OFFSET_X - offsetRight.getX());
+			offsetHealthX = (location.getX() + RESIZED_BOTTOM_OFFSET_X - offsetLeft.getX());
+			offsetPrayerY = (location.getY() - RESIZED_BOTTOM_OFFSET_Y - offsetRight.getY());
+			offsetHealthY = (location.getY() - RESIZED_BOTTOM_OFFSET_Y - offsetRight.getY());
+		}
+		else
+		{
+			offsetPrayerX += curWidget.getWidth();
+		}
+
+		Color healthBar = HEALTH_COLOR;
+		final int poisonState = client.getVar(VarPlayer.IS_POISONED);
+		if (poisonState > 0 && poisonState < 50)
+		{
+			healthBar = POISONED_COLOR;
+		}
+		else if (poisonState >= 1000000)
+		{
+			healthBar = VENOMED_COLOR;
+		}
+
+		final int maxHealth = client.getRealSkillLevel(Skill.HITPOINTS);
+		final int maxPrayer = client.getRealSkillLevel(Skill.PRAYER);
+		final int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
+		final int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+		final int quickPrayerState = client.getVar(Varbits.QUICK_PRAYER);
+		final Color prayerBar = quickPrayerState == 1 ? QUICK_PRAYER_COLOR : PRAYER_COLOR;
+
+		renderBar(g, offsetHealthX, offsetHealthY,
+				maxHealth, currentHealth, height, healthBar);
+
+		renderBar(g, offsetPrayerX, offsetPrayerY,
+				maxPrayer, currentPrayer, height, prayerBar);
+
+		if (config.enableRestorationBars())
+		{
+			int prayerHealValue = 0;
+			int foodHealValue = 0;
+			final MenuEntry[] menu = client.getMenuEntries();
+			final int menuSize = menu.length;
+			final MenuEntry entry = menu[menuSize - 1];
+			final Effect change = itemStatService.getItemStatChanges(entry.getIdentifier());
+
+			if (change != null & entry.getParam1() == WidgetInfo.INVENTORY.getId())
+			{
+				final StatsChanges statsChanges = change.calculate(client);
+				for (final StatChange c : statsChanges.getStatChanges())
+				{
+					if (c.getStat().getName().equals("Hitpoints"))
+					{
+						foodHealValue = Integer.parseInt(c.getTheoretical());
+					}
+
+					if (c.getStat().getName().equals("Prayer"))
+					{
+						prayerHealValue = Integer.parseInt(c.getTheoretical());
```

Comment:
This will not work, e.g for yellow stew, because the values will be 

```
change.setRelative("±" + spiceBoostCapped);
change.setTheoretical("±" + spiceBoost);
```

Reply:
Well, with bars enabled and when you would hover over the yellow stew this will start throwing NumberFormat(Runtime) exceptions

Reply:
It now checks if the entry is a spicy stew, or if you are trying to use a spice on a stew, it threw format exceptions there too.